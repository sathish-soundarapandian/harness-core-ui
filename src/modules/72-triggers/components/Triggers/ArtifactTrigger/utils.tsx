import type { TriggerBaseType, TriggerArtifactType } from '../TriggerInterface'

export interface ArtifactTriggerValues {
  triggerType: TriggerBaseType
  artifactType: TriggerArtifactType
}
