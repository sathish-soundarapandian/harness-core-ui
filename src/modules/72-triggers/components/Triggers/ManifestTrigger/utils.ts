import type { TriggerBaseType, ManifestType } from '../TriggerInterface'

export interface HelmChartInitialValues {
  triggerType: TriggerBaseType
  manifestType: ManifestType
}
