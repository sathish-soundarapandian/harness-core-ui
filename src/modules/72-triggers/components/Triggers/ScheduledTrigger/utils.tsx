/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ObjectSchema, string, TestContext, ValidationError } from 'yup'
import { isEmpty } from 'lodash-es'
import { illegalIdentifiers, regexIdentifier } from '@common/utils/StringUtils'
import { CronFormat, isCronValid } from '@triggers/components/steps/SchedulePanel/components/utils'
import type { StringKeys, UseStringsReturn } from 'framework/strings'
import type { NGTriggerSourceV2, PipelineInfoConfig } from 'services/pipeline-ng'
import type { PanelInterface } from '@triggers/components/TabWizard/TabWizard'
import { NameIdentifierSchema } from '@common/utils/Validation'
import type { ScheduleType, TriggerBaseType } from '../TriggerInterface'

export interface ScheduledInitialValuesInterface {
  triggerType: TriggerBaseType
  scheduleType: ScheduleType
}

export interface FlatInitialValuesInterface {
  triggerType: NGTriggerSourceV2['type']
  identifier?: string
  tags?: {
    [key: string]: string
  }
  pipeline?: string | PipelineInfoConfig
  originalPipeline?: PipelineInfoConfig
  resolvedPipeline?: PipelineInfoConfig
  inputSetTemplateYamlObj?: {
    pipeline: PipelineInfoConfig | Record<string, never>
  }
  name?: string
  selectedScheduleTab?: string
  pipelineBranchName?: string
  inputSetRefs?: string[]
  cronFormat?: CronFormat
}

export interface FlatOnEditValuesInterface {
  name?: string
  identifier?: string
  // targetIdentifier: string
  description?: string
  tags?: {
    [key: string]: string
  }
  pipeline?: PipelineInfoConfig
  triggerType: NGTriggerSourceV2['type']
  manifestType?: string
  artifactType?: string
  originalPipeline?: PipelineInfoConfig
  resolvedPipeline?: PipelineInfoConfig
  selectedScheduleTab?: string
  minutes?: string
  expression?: string
  stageId?: string
  inputSetTemplateYamlObj?: {
    pipeline: PipelineInfoConfig | Record<string, never>
  }
  versionValue?: string
  versionOperator?: string
  buildValue?: string
  buildOperator?: string
  pipelineBranchName?: string
  inputSetRefs?: string[]
  cronFormat?: CronFormat
}

export interface FlatValidScheduleFormikValuesInterface {
  name: string
  identifier: string
  description?: string
  tags?: {
    [key: string]: string
  }
  target?: string
  targetIdentifier?: string
  pipeline: PipelineInfoConfig
  resolvedPipeline?: PipelineInfoConfig
  sourceRepo: string
  triggerType: NGTriggerSourceV2['type']
  expression: string
  pipelineBranchName?: string
  inputSetRefs?: string[]
  stagesToExecute?: string[]
  cronFormat: CronFormat
}

const isIdentifierIllegal = (identifier: string): boolean =>
  regexIdentifier.test(identifier) && illegalIdentifiers.includes(identifier)

const checkValidOverview = ({ formikValues }: { formikValues: { [key: string]: any } }): boolean =>
  isIdentifierIllegal(formikValues?.identifier) ? false : true

const checkValidCronExpression = ({ formikValues }: { formikValues: { [key: string]: any } }): boolean =>
  isCronValid(formikValues?.expression || '', formikValues?.cronFormat === CronFormat.QUARTZ)

const checkValidPipelineInput = ({ formikErrors }: { formikErrors: { [key: string]: any } }): boolean => {
  if (!isEmpty(formikErrors?.pipeline) || !isEmpty(formikErrors?.stages)) {
    return false
  }
  return true
}

export const getPanels = (getString: UseStringsReturn['getString']): PanelInterface[] | [] => {
  return [
    {
      id: 'Trigger Overview',
      tabTitle: getString('overview'),
      checkValidPanel: checkValidOverview,
      requiredFields: ['name', 'identifier'] // conditional required validations checkValidTriggerConfiguration
    },
    {
      id: 'Schedule',
      tabTitle: getString('common.schedule'),
      checkValidPanel: checkValidCronExpression,
      requiredFields: ['expression']
    },
    {
      id: 'Pipeline Input',
      tabTitle: getString('triggers.pipelineInputLabel'),
      checkValidPanel: checkValidPipelineInput
    }
  ]
}
function getCronExpressionValidationError(
  this: TestContext,
  isValidCron: boolean,
  getString: UseStringsReturn['getString']
): boolean | ValidationError {
  if (isValidCron) {
    return true
  } else {
    return this.createError({
      message: getString('triggers.validation.cronExpression')
    })
  }
}

export const getValidationSchema = (
  getString: (key: StringKeys, params?: any) => string
): ObjectSchema<Record<string, any> | undefined> => {
  return NameIdentifierSchema(getString, {
    nameRequiredErrorMsg: getString('triggers.validation.triggerName')
  }).shape({
    expression: string().when('cronFormat', {
      is: val => val === CronFormat.QUARTZ,
      then: string().test({
        test(val: string): boolean | ValidationError {
          const isValidCron = isCronValid(val || '', true)
          return getCronExpressionValidationError.call(this, isValidCron, getString)
        }
      }),
      otherwise: string().test({
        test(val: string): boolean | ValidationError {
          const isValidCron = isCronValid(val || '', false)
          return getCronExpressionValidationError.call(this, isValidCron, getString)
        }
      })
    })
  })
}
