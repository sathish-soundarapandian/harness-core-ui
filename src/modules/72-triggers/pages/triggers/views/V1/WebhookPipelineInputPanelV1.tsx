/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState, useEffect, useMemo, useCallback } from 'react'
import { useParams } from 'react-router-dom'
import { Layout, Text, NestedAccordionProvider, HarnessDocTooltip, PageSpinner } from '@harness/uicore'
import { get, remove } from 'lodash-es'
import { InputSetSelector, InputSetSelectorProps } from '@pipeline/components/InputSetSelector/InputSetSelector'
import { getInputSetForPipelinePromise, InputSetResponse, ResponseInputSetResponse } from 'services/pipeline-ng'
import { useStrings } from 'framework/strings'
import { GitSyncStoreProvider } from 'framework/GitRepoStore/GitSyncStoreContext'
import { useQueryParams } from '@common/hooks'
import type { GitQueryParams, PipelinePathProps } from '@common/interfaces/RouteInterfaces'
import type { InputSetValue } from '@pipeline/components/InputSetSelector/utils'

import NewInputSetModal from '@pipeline/components/InputSetForm/NewInputSetModal'
import { useInputSetsV1 } from '@pipeline/v1/components/RunPipelineModalV1/useInputSetsV1'
import { getTriggerInputSetsBranchQueryParameter } from '@triggers/components/Triggers/utils'
import { getErrorMessage } from '../../utils/TriggersWizardPageUtils'
import css from '../WebhookPipelineInputPanel.module.scss'

interface WebhookPipelineInputPanelPropsInterface {
  formikProps?: any
  gitAwareForTriggerEnabled?: boolean
}

function WebhookPipelineInputPanelFormV1({
  formikProps,
  gitAwareForTriggerEnabled
}: WebhookPipelineInputPanelPropsInterface): React.ReactElement {
  const {
    values: { inputSetSelected, pipeline }
  } = formikProps

  const { getString } = useStrings()
  const { repoIdentifier, branch, connectorRef, repoName, storeType } = useQueryParams<GitQueryParams>()
  const [selectedInputSets, setSelectedInputSets] = useState<InputSetSelectorProps['value']>(inputSetSelected)
  const [invalidInputSetIds, setInvalidInputSetIds] = useState<Array<string>>([])

  const { orgIdentifier, accountId, projectIdentifier, pipelineIdentifier } = useParams<PipelinePathProps>()

  const {
    hasRuntimeInputs,
    hasCodebaseInputs,
    isLoading: loadingInputSets
  } = useInputSetsV1({
    accountId,
    projectIdentifier,
    orgIdentifier,
    pipelineIdentifier,
    branch,
    repoIdentifier,
    connectorRef
  })

  const inputSetSelectedBranch = useMemo(() => {
    return getTriggerInputSetsBranchQueryParameter({
      gitAwareForTriggerEnabled,
      pipelineBranchName: formikProps?.values?.pipelineBranchName,
      branch
    })
  }, [gitAwareForTriggerEnabled, branch, formikProps?.values?.pipelineBranchName])

  const onReconcile = (inpSetId: string): void => {
    remove(invalidInputSetIds, id => id === inpSetId)
    setInvalidInputSetIds(invalidInputSetIds)
  }

  const inputSetQueryParams = useMemo(
    () => ({
      accountIdentifier: accountId,
      orgIdentifier,
      pipelineIdentifier,
      projectIdentifier,
      repoIdentifier,
      connectorRef,
      repoName,
      storeType,
      branch: getTriggerInputSetsBranchQueryParameter({
        gitAwareForTriggerEnabled,
        pipelineBranchName: formikProps?.values?.pipelineBranchName,
        branch
      })
    }),
    [
      accountId,
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      repoIdentifier,
      formikProps?.values?.pipelineBranchName,
      connectorRef,
      repoName,
      storeType,
      branch,
      gitAwareForTriggerEnabled
    ]
  )

  useEffect(() => {
    setSelectedInputSets(inputSetSelected)
  }, [inputSetSelected])

  const [fetchInputSetsInProgress, setFetchInputSetsInProgress] = useState(false)
  const [inputSetError, setInputSetError] = useState('')

  useEffect(
    function fetchInputSetsFromInputSetRefs() {
      async function fetchInputSets(): Promise<void> {
        setInputSetError('')

        const inputSetRefs = formikProps?.values?.inputSetRefs
        const inputSetRefsLength = formikProps?.values?.inputSetRefs?.length
        const selectedInputSetsLength = selectedInputSets?.length

        if (
          inputSetRefsLength &&
          selectedInputSetsLength &&
          inputSetRefsLength === selectedInputSetsLength &&
          inputSetRefs?.every((ref: string) => selectedInputSets?.find(item => item.value === ref))
        ) {
          // No need to fetch input sets if they are fetched already
          return
        }

        Promise.all(
          inputSetRefs.map(async (inputSetIdentifier: string): Promise<any> => {
            const data = await getInputSetForPipelinePromise({
              inputSetIdentifier,
              queryParams: inputSetQueryParams
            })

            return data
          })
        )
          .then(results => {
            const error = (results || []).find(result => get(result, 'status') === 'ERROR')
            if (error) {
              if (!inputSetSelected) {
                formikProps.setValues({
                  ...formikProps.values,
                  inputSetSelected: []
                })
              }
              setInputSetError(getErrorMessage(error))
            } else if (results?.length) {
              const inputSets = (results as unknown as { data: InputSetResponse }[]).map(
                ({ data: { identifier, name, gitDetails } }) => ({
                  label: name,
                  value: identifier,
                  type: 'INPUT_SET',
                  gitDetails
                })
              )

              setSelectedInputSets(inputSets as InputSetValue[])
            }
          })
          .catch(exception => {
            setInputSetError(getErrorMessage(exception))
          })
          .finally(() => {
            setFetchInputSetsInProgress(false)
          })
      }

      if (!fetchInputSetsInProgress && !inputSetSelected && formikProps?.values?.inputSetRefs?.length) {
        setFetchInputSetsInProgress(true)
        fetchInputSets()
      }
    },
    [
      formikProps?.values?.inputSetRefs,
      inputSetSelected,
      inputSetQueryParams,
      fetchInputSetsInProgress,
      selectedInputSets,
      formikProps
    ]
  )

  const showPipelineInputSetSelector = useMemo(
    () => hasRuntimeInputs || hasCodebaseInputs,
    [pipeline, hasRuntimeInputs, hasCodebaseInputs]
  )

  const [showNewInputSetModal, setShowNewInputSetModal] = useState(false)
  const onNewInputSetSuccess = useCallback(
    (response: ResponseInputSetResponse) => {
      const inputSet = response.data as InputSetResponse
      const _inputSetSelected = (selectedInputSets || []).concat({
        label: inputSet.name as string,
        value: inputSet.identifier as string,
        type: 'INPUT_SET',
        gitDetails: inputSet.gitDetails
      })

      setInputSetError('')
      setSelectedInputSets(_inputSetSelected)

      formikProps.setValues({
        ...formikProps.values,
        inputSetSelected: _inputSetSelected,
        inputSetRefs: _inputSetSelected.map(_inputSet => _inputSet.value)
      })
    },
    [formikProps, selectedInputSets]
  )

  useEffect(() => {
    setInputSetError(formikProps?.errors?.inputSetRefs)
  }, [setInputSetError, formikProps?.errors?.inputSetRefs])

  // Don't show spinner when fetching is triggered by typing from
  // Pipeline Reference. Giving users a better experience
  const isPipelineBranchNameInFocus = (): boolean =>
    !!gitAwareForTriggerEnabled &&
    !!document.activeElement &&
    document.activeElement === document.querySelector('input[name="pipelineBranchName"]')

  return (
    <Layout.Vertical className={css.webhookPipelineInputContainer} spacing="large" padding="none">
      {loadingInputSets && !isPipelineBranchNameInFocus() ? (
        <div style={{ position: 'relative', height: 'calc(100vh - 128px)' }}>
          <PageSpinner />
        </div>
      ) : hasRuntimeInputs || hasCodebaseInputs || gitAwareForTriggerEnabled ? (
        <div className={css.inputsetGrid}>
          <div className={css.inputSetContent}>
            {showPipelineInputSetSelector ? (
              <div className={css.pipelineInputRow}>
                <Text className={css.formContentTitle} inline={true} data-tooltip-id="pipelineInputLabel">
                  {getString('triggers.pipelineInputLabel')}
                  <HarnessDocTooltip tooltipId="pipelineInputLabel" useStandAlone={true} />
                </Text>

                <GitSyncStoreProvider>
                  <InputSetSelector
                    pipelineIdentifier={pipelineIdentifier}
                    onChange={value => {
                      setInputSetError('')
                      setSelectedInputSets(value)
                      formikProps.setValues({
                        ...formikProps.values,
                        inputSetRefs: (value || []).map(v => v.value),
                        inputSetSelected: value
                      })
                    }}
                    value={selectedInputSets}
                    selectedValueClass={css.inputSetSelectedValue}
                    selectedRepo={gitAwareForTriggerEnabled ? repoName : repoIdentifier}
                    selectedBranch={inputSetSelectedBranch}
                    showNewInputSet={true}
                    onNewInputSetClick={() => setShowNewInputSetModal(true)}
                    invalidInputSetReferences={invalidInputSetIds}
                    onReconcile={onReconcile}
                  />
                </GitSyncStoreProvider>
                {inputSetError ? <Text intent="danger">{inputSetError}</Text> : null}
                <div className={css.divider} />
                {showNewInputSetModal && (
                  <NewInputSetModal
                    isModalOpen={showNewInputSetModal}
                    closeModal={() => setShowNewInputSetModal(false)}
                    onCreateSuccess={onNewInputSetSuccess}
                    isSimplifiedYAML
                  />
                )}
              </div>
            ) : null}
          </div>
        </div>
      ) : (
        <Layout.Vertical padding={{ left: 'small', right: 'small' }} margin="large" spacing="large">
          <Text className={css.formContentTitle} inline={true} tooltipProps={{ dataTooltipId: 'pipelineInputLabel' }}>
            {getString('triggers.pipelineInputLabel')}
          </Text>
          <Layout.Vertical className={css.formContent}>
            <Text>{getString('pipeline.pipelineInputPanel.noRuntimeInputs')}</Text>
          </Layout.Vertical>
        </Layout.Vertical>
      )}
    </Layout.Vertical>
  )
}

const WebhookPipelineInputPanelV1: React.FC<WebhookPipelineInputPanelPropsInterface> = props => {
  return (
    <NestedAccordionProvider>
      <WebhookPipelineInputPanelFormV1 {...props} />
    </NestedAccordionProvider>
  )
}
export default WebhookPipelineInputPanelV1
