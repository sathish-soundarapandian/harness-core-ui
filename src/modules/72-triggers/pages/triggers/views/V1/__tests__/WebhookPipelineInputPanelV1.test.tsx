/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { Formik, FormikForm } from '@harness/uicore'
import { renderHook } from '@testing-library/react-hooks'
import { useGetPipelineInputsQuery } from '@harnessio/react-pipeline-service-client'
import { PipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { TestWrapper } from '@common/utils/testUtils'
import { useStrings } from 'framework/strings'
import { branchStatusMock, gitConfigs, sourceCodeManagers } from '@connectors/mocks/mock'
import {
  ConnectorResponse,
  GetEnvironmentList,
  pipelineInputInitialValues,
  getTriggerConfigDefaultProps
} from './mocks'

import WebhookPipelineInputPanelV1 from '../WebhookPipelineInputPanelV1'
jest.mock('@common/components/YAMLBuilder/YamlBuilder')

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),

  useMutateAsGet: jest.fn()
}))

const getListOfBranchesWithStatus = jest.fn(() => Promise.resolve(branchStatusMock))
const getListGitSync = jest.fn(() => Promise.resolve(gitConfigs))
jest.mock('@harnessio/react-pipeline-service-client', () => ({
  useGetPipelineInputsQuery: jest.fn(() => ({}))
}))
jest.mock('services/cd-ng', () => ({
  useGetEnvironmentAccessList: jest.fn(() => GetEnvironmentList),
  useGetConnector: jest.fn(() => ConnectorResponse),
  useGetListOfBranchesWithStatus: jest.fn().mockImplementation(() => {
    return { data: branchStatusMock, refetch: getListOfBranchesWithStatus, loading: false }
  }),
  useListGitSync: jest.fn().mockImplementation(() => {
    return { data: gitConfigs, refetch: getListGitSync }
  })
}))

jest.mock('services/cd-ng-rq', () => ({
  useGetSourceCodeManagersQuery: jest.fn().mockImplementation(() => {
    return { data: sourceCodeManagers, refetch: jest.fn() }
  })
}))

const defaultTriggerConfigDefaultProps = getTriggerConfigDefaultProps({})

const wrapper = ({ children }: React.PropsWithChildren<unknown>): React.ReactElement => (
  <TestWrapper>{children}</TestWrapper>
)
const { result } = renderHook(() => useStrings(), { wrapper })

function WrapperComponent(): JSX.Element {
  return (
    <TestWrapper
      path="/account/:accountId/org/:orgIdentifier/project/:projectIdentifier"
      pathParams={{
        projectIdentifier: 'projectIdentifier',
        orgIdentifier: 'orgIdentifier',
        accountId: 'accountId'
      }}
    >
      <Formik initialValues={pipelineInputInitialValues} onSubmit={() => undefined} formName="wrapperComponentTestForm">
        {formikProps => (
          <FormikForm>
            <PipelineContext.Provider
              value={
                {
                  state: { pipeline: { name: '', identifier: '' } } as any,
                  getStageFromPipeline: jest.fn((_stageId, pipeline) => ({
                    stage: pipeline.stages[0],
                    parent: undefined
                  }))
                } as any
              }
            >
              <WebhookPipelineInputPanelV1 {...defaultTriggerConfigDefaultProps} formikProps={formikProps} />
            </PipelineContext.Provider>
          </FormikForm>
        )}
      </Formik>
    </TestWrapper>
  )
}

describe('WebhookPipelineInputPanelV1 Triggers tests', () => {
  describe('Renders/snapshots', () => {
    test('Initial Render - Pipeline Input Panel with no inputs', async () => {
      const { container } = render(<WrapperComponent />)
      await waitFor(() => expect(result.current.getString('triggers.pipelineInputLabel')).toBeTruthy())
      expect(container).toMatchSnapshot()
    })

    test('Initial Render - Pipeline Input Panel with CI Codebase inputs', async () => {
      const mockCodebaseInputs = {
        data: {
          repository: {
            reference: 'branch',
            value: ''
          }
        },
        status: 'SUCCESS'
      }
      ;(useGetPipelineInputsQuery as jest.Mock).mockImplementation().mockReturnValue(mockCodebaseInputs)
      const { container } = render(<WrapperComponent />)
      await waitFor(() => expect(result.current.getString('triggers.pipelineInputLabel')).toBeTruthy())
      await waitFor(() => expect(result.current.getString('ciCodebase')).toBeTruthy())
      expect(container).toMatchSnapshot()
    })

    test('Initial Render - Pipeline Input Panel with runtime inputs', async () => {
      const mockCodebaseInputs = {
        inputs: {
          image: {
            prompt: false,
            required: true,
            default: 'golang',
            type: 'string',
            desc: 'image name'
          },
          repo: {
            prompt: true,
            required: true,
            type: 'string',
            desc: 'repository name'
          }
        },
        repository: {
          reference: {
            type: {
              prompt: false,
              required: true,
              type: 'string',
              enums: ['branch', 'tag', 'pr']
            },
            value: {
              prompt: false,
              required: true,
              type: 'string'
            }
          }
        }
      }
      ;(useGetPipelineInputsQuery as jest.Mock).mockImplementation().mockReturnValue(mockCodebaseInputs)
      const { container } = render(<WrapperComponent />)
      await waitFor(() => expect(result.current.getString('connectors.parameters')).toBeTruthy())
      expect(container).toMatchSnapshot()
    })
  })
})
