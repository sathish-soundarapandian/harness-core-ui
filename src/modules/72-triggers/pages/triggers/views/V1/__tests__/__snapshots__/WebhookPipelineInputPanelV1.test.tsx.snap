// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhookPipelineInputPanelV1 Triggers tests Renders/snapshots Initial Render - Pipeline Input Panel with CI Codebase inputs 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <div
          class="Layout--vertical Layout--layout-spacing-large StyledProps--main webhookPipelineInputContainer StyledProps--padding-none"
        >
          <div
            class="inputsetGrid"
          >
            <div
              class="inputSetContent"
            >
              <div
                class="pipelineInputRow"
              >
                <span
                  class="StyledProps--font StyledProps--main formContentTitle StyledProps--inline"
                  data-tooltip-id="pipelineInputLabel"
                >
                  triggers.pipelineInputLabel
                </span>
                <span
                  class="bp3-popover-wrapper isPopoverParent"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                    >
                      <button
                        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main addInputSetButton StyledProps--color StyledProps--color-primary7 Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon Button--variation Button--variation-link"
                        type="button"
                      >
                        <span
                          class="bp3-icon bp3-icon-small-plus StyledProps--main"
                          icon="small-plus"
                        >
                          <svg
                            data-icon="small-plus"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <desc>
                              small-plus
                            </desc>
                            <path
                              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </span>
                        <span
                          class="bp3-button-text"
                        >
                          pipeline.inputSets.createNewInputSet
                        </span>
                      </button>
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--font-variation-form-label StyledProps--color StyledProps--color-grey200"
                        style="align-self: center;"
                      >
                        |
                      </p>
                      <button
                        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main addInputSetButton StyledProps--color StyledProps--color-primary7 Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon Button--variation Button--variation-link"
                        type="button"
                      >
                        <span
                          class="bp3-icon bp3-icon-small-plus StyledProps--main"
                          icon="small-plus"
                        >
                          <svg
                            data-icon="small-plus"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <desc>
                              small-plus
                            </desc>
                            <path
                              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </span>
                        <span
                          class="bp3-button-text"
                        >
                          pipeline.inputSets.selectPlaceholder
                        </span>
                      </button>
                    </div>
                    <div
                      class="renderSelectedValue inputSetSelectedValue"
                    />
                  </span>
                </span>
                <div
                  class="divider"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`WebhookPipelineInputPanelV1 Triggers tests Renders/snapshots Initial Render - Pipeline Input Panel with no inputs 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <div
          class="Layout--vertical Layout--layout-spacing-large StyledProps--main webhookPipelineInputContainer StyledProps--padding-none"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-large StyledProps--main StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--margin-large"
          >
            <div
              class="Text--withDocsTooltip"
              data-tooltip-id="pipelineInputLabel"
            >
              <span
                class="StyledProps--font StyledProps--main formContentTitle StyledProps--inline"
              >
                triggers.pipelineInputLabel
              </span>
            </div>
            <div
              class="Layout--vertical StyledProps--main formContent"
            >
              <p
                class="StyledProps--font StyledProps--main"
              >
                pipeline.pipelineInputPanel.noRuntimeInputs
              </p>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`WebhookPipelineInputPanelV1 Triggers tests Renders/snapshots Initial Render - Pipeline Input Panel with runtime inputs 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <div
          class="Layout--vertical Layout--layout-spacing-large StyledProps--main webhookPipelineInputContainer StyledProps--padding-none"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-large StyledProps--main StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--margin-large"
          >
            <div
              class="Text--withDocsTooltip"
              data-tooltip-id="pipelineInputLabel"
            >
              <span
                class="StyledProps--font StyledProps--main formContentTitle StyledProps--inline"
              >
                triggers.pipelineInputLabel
              </span>
            </div>
            <div
              class="Layout--vertical StyledProps--main formContent"
            >
              <p
                class="StyledProps--font StyledProps--main"
              >
                pipeline.pipelineInputPanel.noRuntimeInputs
              </p>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
